src/lolex-src.js

```diff
@@ -531,6 +534,30 @@ function withGlobal(_global) {
             return 16 - ((clock.now - start) % 16);
         }
 
+        function hrtime(prev) {
+            var millisSinceStart = clock.now - adjustedSystemTime[0] - start;
+            var secsSinceStart = Math.floor( millisSinceStart / 1000);
+            var remainderInNanos = (millisSinceStart - secsSinceStart * 1e3 ) * 1e6 + nanos - adjustedSystemTime[1];
+
+            if (Array.isArray(prev)) {
+                if ( prev[1] > 1e9 ) {
+                    throw new TypeError("Number of nanoseconds can't exceed a billion");
+                }
+
+                var oldSecs = prev[0];
+                var nanoDiff = remainderInNanos - prev[1];
+                var secDiff = secsSinceStart - oldSecs;
+
+                if (nanoDiff < 0) {
+                    nanoDiff += 1 * 1e9;
```

Comment:
why the `1`?

Reply:
Do you mean "Why do I add a billion" or why do I use `1 * 1e9` instead of just `1e9`. In case of the latter, no particular reason. I can change it.

Reply:
The latter 🙂